### PR TITLE
Feat/add lottie animation

### PR DESCRIPTION
**Pull request by mmzek**
lottie-animation
lottie-animation1

When error occurres or SKS's closed, the lottie animation "closed" is showing on SKS Menu.
In my opinion, the offline menu functionality is out of scope for this task and should be addressed as a separate task.

### CODE REVIEW

**Comment by mikolaj-jalocha** [comment]
It appears that again some weird linter error occurred. Imo this happens only if build_runner has not been sent before push. 
Linter complaints should disapear once merged into main.

File: android/app/build.gradle [hunk 1]
```diff
@@ -35,8 +35,8 @@ android {
     ndkVersion flutter.ndkVersion
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
```
Comment (tomasz-trela) [inline android/app/build.gradle:39]: Why did you change it?

File: android/gradle.properties [hunk 2]
```diff
@@ -1,3 +1,5 @@
 org.gradle.jvmargs=-Xmx4G
 android.useAndroidX=true
 android.enableJetifier=true
+org.gradle.java.home=C:/Program Files/Eclipse Adoptium/jdk-17.0.13.11-hotspot
```
Comment (tomasz-trela) [inline android/gradle.properties:4]: image
why did you add this java path?

**Review by tomasz-trela — COMMENTED** [review]
I don't think it's good to make these changes (gradle properties and java version)

File: android/app/build.gradle (hunk 1 above)
Comment (mikolaj-jalocha) [reply android/app/build.gradle:39]: Probably this was needed  because of some demands of Android Studio. My advice is to make such changes only locally (at least for now, we can discuss global change of java version ) and don't push them on github. 

Please change them back to 1.8

File: android/gradle.properties (hunk 2 above)
Comment (mikolaj-jalocha) [reply android/gradle.properties:4]: Same here,  we can ommit this file when pushing chagnes on remote. 

Please change it back

File: assets/animations/closed.json
Comment (mikolaj-jalocha) [inline assets/animations/closed.json:1]: I like the anim :-)

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import "package:flutter/material.dart";
+import "package:lottie/lottie.dart";
+
+import "../../../../../theme/app_theme.dart";
+import "../../../../../utils/context_extensions.dart";
+import "../../../../gen/assets.gen.dart";
+
+class LottieLoading extends StatefulWidget {
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:8]: I'd think about better naming, that reflects where such lottie animation is used, eg. "SKSMenuLottieAnimation" or sth like that.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  @override
+  _LottieLoadingState createState() => _LottieLoadingState();
+}
+
+class _LottieLoadingState extends State<LottieLoading> {
+  final bool _showOldMenu = false;
+
+  void _gotoOldMenu() {
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:20]: Some debugging tools i suposse :-) Should be deleted

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+class LottieLoading extends StatefulWidget {
+  const LottieLoading({
+    super.key,
+  });
+
+  @override
+  _LottieLoadingState createState() => _LottieLoadingState();
+}
+
+class _LottieLoadingState extends State<LottieLoading> {
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:17]: Actually, I think that we don;t need here state class.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    return Scaffold(
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          const Spacer(),
+          Center(
+            child: SizedBox(
+              width: 200,
+              height: 200,
+              child: Lottie.asset(
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:38]: Lottie here could be a little more customized. Ie. cache, frame rate. Please see reference in file used somewhere else: 

image

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                Assets.animations.closed,
+                repeat: false,
+              ),
+            ),
+          ),
+          const Spacer(),
+          Align(
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:45]: As this tasks's scope was reduced, we don;t need such a button here. 

But it would be great if some message's displayed when error occured. Such us, menu is not available, pls try again later, or sth like that. So that user could understand what's going on

File: pubspec.yaml [hunk 8]
```diff
@@ -149,6 +149,7 @@ flutter:
     - assets/animations/search.json
     - assets/animations/offline.json
     - assets/animations/bug.json
+    - assets/animations/closed.json
```
Comment (mikolaj-jalocha) [inline pubspec.yaml:152]: WDYT about a little more descriptive name: sks_closed.json ?

**Review by mikolaj-jalocha — CHANGES_REQUESTED** [review]
First of all, congrats of your first commit here and working code!  Animation looks great😎

I left few suggestions of mine regarding UI. I think button could be removed (as of now) and some text regarding error could be displayed as well. 

I also left notes regarding better use of lottie and some regarding changes of files that should only be made locally.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Column(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          const Spacer(),
+          Center(
+            child: SizedBox(
+              width: 200,
+              height: 200,
```
Comment (tomasz-trela) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:37]: Some values can be separated to the ui_config as costant values.

**Review by tomasz-trela — CHANGES_REQUESTED** [review]
We don't really want to have magic nubers in code.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart (hunk 9 above)
Comment (simon-the-shark) [reply lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:37]: Actually I would reccomend using `SizedBox.square` constructor which removes duplication and config file extraction can prolly be skipped then.

**Review by simon-the-shark — COMMENTED** [review]
So good job and congrats for the first completed task.

I've left one note about the sized box, but everything else was pretty much covered by Tomek and Mikołaj, so good job guys too.

As for java build settings, let's not change anything globally for now. I've actually seen that Andrea has made some fancy tool related on the topic, so someone can investigate on that: https://gist.github.com/bizz84/605e2ca2088cb4acb7a076ca993f41cd?trk=public_post_comment-text But let's rollback for now

**Comment by mmzek** [comment]
Sorry for pushing java path etc., @mikolaj-jalocha was right (some demands of Android Studio), will not happen again. 
I applied changes you mentioned in review and added text describing whats wrong on the screen. Still have some problems with linter even tho flutter analyze shows no issues found.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:1]: Name of the file could be different since this animation's not responsible for loading. In fact we could use that class inside top level sks_menu_screen as a private class.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+              child: Lottie.asset(
+                Assets.animations.sksClosed,
+                fit: BoxFit.cover,
+                repeat: false,
+                frameRate: const FrameRate(LottieAnimationConfig.frameRate),
+                renderCache: RenderCache.drawingCommands,
+              ),
+            ),
+          Align(
+              child: Text(
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:31]: Maybe it'd be good to add  detailed error message (error message that comes from api). Maybe in small print and in different color.

File: lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+import "../../../../config/ui_config.dart";
+import "../../../../gen/assets.gen.dart";
+
+class SKSMenuLottieAnimation extends StatelessWidget {
+  const SKSMenuLottieAnimation({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/widgets/sks_menu_loading.dart:15]: It'd be great to add logger somewhere here:

`Logger().e(error.toString());`

File: lib/features/sks-menu/presentation/sks_menu_screen.dart [hunk 12]
```diff
@@ -63,11 +56,7 @@ class _SksMenuView extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     if (sksMenuData.meals.isEmpty) {
-      return const Scaffold(
-        body: Center(
-          child: CircularProgressIndicator(),
-        ),
-      );
+      return const SKSMenuLottieAnimation();
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/sks_menu_screen.dart:59]: Here we should check if menu's online or not and then show animation (list of meals now is only empty in case of error, if sks's closed cached data are provided, so this condition'll never work)

**Review by mikolaj-jalocha — CHANGES_REQUESTED** [review]
Good job! Code looks cleaner now. I tested it and animations works good on my device.
I left few notes (one is crucial, pls see what happens if menu is offline)

File: lib/features/sks-menu/presentation/sks_menu_screen.dart [hunk 13]
```diff
@@ -63,11 +56,7 @@ class _SksMenuView extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     if (sksMenuData.meals.isEmpty) {
```
Comment (mikolaj-jalocha) [inline lib/features/sks-menu/presentation/sks_menu_screen.dart:58]: Here we should check if menu's online or not and then show animation (list of meals now is only empty in case of error, if sks's closed cached data are provided, so this condition'll never work)

**Review by mikolaj-jalocha — CHANGES_REQUESTED** [review]
and one more

**Review by mikolaj-jalocha — DISMISSED** [review]
LGTM! 🚀

**Review by simon-the-shark — APPROVED** [review]
Looks good to me. I would maybe think of changing the animation color, but that's secondary consern. Merging it now. Good job!
